### PR TITLE
Add offset to link and collision pose

### PR DESCRIPTION
**Pull request by claireyywang**
Use `pose` represent relative transformation to parent entity. `GetWorldPose` represents pose relative to world. This PR takes account of link and collision relative pose to compute their worldPose.  

This is the first step to implement #57

### CODE REVIEW

File: tpe/lib/src/Link.hh [hunk 1]
```diff
@@ -42,6 +44,22 @@ class IGNITION_PHYSICS_TPELIB_VISIBLE Link : public Entity
...
+  /// \brief Get the relative transform to its parent model
+  /// \return relative transform
+  public: const Eigen::Isometry3d GetTf();
+
+  /// \brief Set the relative transform to its parent model
+  /// \param[in] _tf relative transform
+  public: void SetTf(const Eigen::Isometry3d _tf);
+
+  /// \brief Private relative transform to its parent entity
+  public: Eigen::Isometry3d tf =
```
Comment (scpeters) [inline tpe/lib/src/Link.hh:61]: according to the comment, should this be `private:`?

**Comment by iche033** [comment]
The changes look good to me. Is this PR still a draft?

**Comment by claireyywang** [comment]
@iche033 awesome, thanks for looking into it. I think this PR is ready to go. I have updates in tpeplugin but that is on a separate branch and doesn't interfere with this one.

File: tpe/lib/src/Collision.hh [hunk 2]
```diff
@@ -18,7 +18,7 @@
 #ifndef IGNITION_PHYSICS_TPE_LIB_SRC_COLLISION_HH_
 #define IGNITION_PHYSICS_TPE_LIB_SRC_COLLISION_HH_
 
-#include <ignition/math/AxisAlignedBox.hh>
+#include <ignition/math/eigen3/Conversions.hh>
```
Comment (iche033) [inline tpe/lib/src/Collision.hh:21]: I think we don't need this include now right?

File: tpe/lib/src/Link.hh [hunk 3]
```diff
@@ -18,6 +18,8 @@
 #ifndef IGNITION_PHYSICS_TPE_LIB_SRC_LINK_HH_
 #define IGNITION_PHYSICS_TPE_LIB_SRC_LINK_HH_
 
+#include <ignition/math/eigen3/Conversions.hh>
```
Comment (iche033) [inline tpe/lib/src/Link.hh:21]: no longer needed?

**Pull request merged**